### ISSUE TITLE
[BUG][HIGH] ZAYA1-8B .1bp GPU decode = deterministic garbage — single-token path drifts (prefill top-1 matches) — confirms #1521's unconfirmed wq/wk/wv1/wv2 transpose note

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#1603]
## Summary

`zaya` with any `ZAYA1-8B.1bp` on the HIP backend generates deterministic token garbage. Prefill logits are essentially correct (top-1 matches the torch reference); **decode drifts from step 1-2 onward** and the hypersensitive MoE router flips → output is byte-level junk (`(xiacached>((cached('$(('$('>xiastigGLOBALS...`). This is the exact failure mode #1521 flagged as unconfirmed: *"Single-token decode reads `wq/wk/wv1/wv2` transposed vs how they're uploaded — if 74B output is garbage, transpose those four."*

## Repro (strix, 2026-08-12, stock kernel 7.0.0-29, amdxdna 0.16.0 OOT)

```
./build/1bit zaya --port 8093 --model models/ZAYA1-8B-fresh.1bp
curl :8093/v1/chat/completions -d '{"messages":[{"role":"user","content":"What is 17*23?"}]}'
# → (xiacached>((cached('$(('$('>xiastigGLOBALSuseridimuthlisievaluation...
```

Runs clean (no HIP error, 8.6 tok/s, 96/96 tokens) — the engine is happy with itself; only the content is wrong.

## Evidence

- **Fresh `.1bp` from the current converter** (built Aug 9, post-#1521): `gguf_to_onebp ZAYA1-8B-Q8_0.gguf ZAYA1-8B-fresh.1bp` → byte-identical garbage to the Aug 8 file. Old file, new file, same junk. Not a stale-artifact issue.
- **Tokenizer verified correct**: fresh `.htok` from the official `tokenizer.json` (262147 tokens); prompt tokenizes right (33 tokens for the sentence). Output garbage is independent of tokenizer.
- **Engine's own coherence probe catches it elsewhere**: with the F32 GGUF, the backend cascade logs `NPU FLM: coherence probe FAILED (degenerate output)` and `CPU (scalar): coherence probe FAILED` — the probe detects this class of degradation for FLM/CPU but the HIP path loads the `.1bp` without a probe.
- **Drift numbers (wiki, obs-2026-08-11)**: prefill `cos 0.926` F32-vs-F32 with top-1 MATCH; decode step1 `0.935` then drifts to `0.54–0.85` as router selections flip. That is NOT arithmetic-order noise (would be cos ~0.9999) — it is a decode-path defect. Prefill-correct/decode-drifts is the signature of the single-token path reading Q/K/V (or the CCA latent KV) in the wrong orientation.
- GGUF path is not a workaround: the Q8_0 GGUF download is internally inconsistent (header vocab 262272 vs `token_embd` 262147 — llama.cpp rejects it), and the in-engine Universal/GGML-Vulkan backend refuses `arch=zaya` MoE layouts (#1520).

## What's ruled out
- Weights: official `Zyphra/ZAYA1-8B` safetensors verified on disk; F16/F32 GGUFs (own converter) load in the ggml fork.
- Tokenizer: fixed (above).
- Kernel/GPU: pure HIP compute works (no launch failures; other kernels correct); this box's GPU path is otherwise healthy.
- `.1bp` layout fix #1521: applied in the converter binary used; garbage persists.

## Suggested fix direction
Per #1521 step 3: transpose `wq`/`wk`/`wv1`/`wv2` in the single-token decode read (one line each) and/or reconcile the CCA latent-KV decode orientation — then run the coherence probe on the HIP path (it currently bypasses it). The probe that catches FLM/CPU degeneration should gate HIP too.

## Ref
- #1521 (unvalidated engine path — its step 4 note is now confirmed by repro)
- #1527 (loader zero-fill tolerances — fixed, but same garbage class)
- Wiki: obs-2026-08-11-zaya-decode-drift-root-caused-f32-noise-amplification (the "F32 arithmetic order noise" conclusion there is contradicted by the cos numbers; re-flagging as decode-path bug)

**Comment by bong-water-water-bong** [comment]
Resolved by #1624 (commit 7ddf91d8): ZAYA1-8B is a hybrid SSM+attention model (cca_*/ssm_conv1d/res_scale tensors, no ffn_gate/up/down). The HIP backend only implements the dense GQA transformer path — the 'deterministic garbage' was this unsupported architecture producing garbage instead of crashing. The backend now rejects ZAYA at init with a clear message. The #1521 transpose note concerns the NPU engine (different code path) and stays tracked there.
